api/service/projects_service.go

```diff
@@ -5,22 +5,31 @@ import (
 	"fmt"
 	"strings"
 
+	"bytes"
+	"encoding/json"
+	"html/template"
+	"net/http"
+
 	"golang.org/x/exp/slices"
 
 	"github.com/pkg/errors"
 
 	"github.com/caraml-dev/mlp/api/repository"
 
+	"github.com/caraml-dev/mlp/api/config"
 	"github.com/caraml-dev/mlp/api/models"
 	"github.com/caraml-dev/mlp/api/pkg/authz/enforcer"
 )
 
 type ProjectsService interface {
 	ListProjects(ctx context.Context, name string, user string) ([]*models.Project, error)
 	CreateProject(ctx context.Context, project *models.Project) (*models.Project, error)
-	UpdateProject(ctx context.Context, project *models.Project) (*models.Project, error)
+	UpdateProject(ctx context.Context, project *models.Project) (*models.Project, string, error)
 	FindByID(projectID models.ID) (*models.Project, error)
 	FindByName(projectName string) (*models.Project, error)
+	MakeRequestPayload(project *models.Project, templateString string) (string, error)
```

Comment:
why are these expose as the interface? These are only used internally by `UpdateProject` right?

Are there any use case for these function to be used outside of the create/update project?